File: src/test/java/com/example/retailstore/strategy/CustomerDiscountStrategyTest.java
package com.example.retailstore.strategy;

import com.example.retailstore.model.Bill;
import org.junit.jupiter.api.Test;

import static com.example.retailstore.model.UserType.CUSTOMER_FOR_TWO_YEARS;
import static com.example.retailstore.model.UserType.EMPLOYEE;
import static org.junit.jupiter.api.Assertions.assertEquals;

class CustomerDiscountStrategyTest {
    @Test
    void shouldCalculateNetPayableAmountWithNoGroceries() {
        Bill bill = new Bill(1000, false, CUSTOMER_FOR_TWO_YEARS);

        CustomerDiscountStrategy strategy = new CustomerDiscountStrategy();

        double actual = strategy.calculateNetPayableAmount(bill);

        assertEquals(900, actual);
    }
}

File: src/test/java/com/example/retailstore/strategy/CustomerDiscountStrategyTest.kt
package com.example.retailstore.strategy

import com.example.retailstore.model.Bill
import com.example.retailstore.model.UserType
import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test


internal class CustomerDiscountStrategyTest {
    @Test
    fun shouldCalculateNetPayableAmountWithNoGroceries() {
        val bill = Bill(1000.0, false, UserType.CUSTOMER_FOR_TWO_YEARS)
        val strategy = CustomerDiscountStrategy()
        val actual = strategy.calculateNetPayableAmount(bill)
        Assertions.assertEquals(900.0, actual)
    }
}